### ISSUE TITLE
When regening walls, they disappear

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#3769]
I've used @brunoperdigao's [Duplicate Aggregate](https://github.com/IfcOpenShell/IfcOpenShell/pull/3328) functionality to duplicate a housing module throughout a site. 

As you can see from the following video, If I 'shift-G' the walls in the original 'parent', they disappear.

file: 

https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/a5e0bfc8-96bc-487d-9781-e98c6823f5db

**Comment by brunoperdigao** [comment]
I did a quit test and I figure that this doesn't happen with these internal walls:
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/57102715/fa263503-9744-4c08-9c6c-69d172562942)

They seem to be the only ones without a boolean relationship, maybe it is related to that? I don't know if it has any relation to the aggregate. I'll try to take a closer look tomorrow.

**Comment by theoryshaw** [comment]
I did a simple test.  As you can see the walls on one aggregate are attached to the slab in both aggregates.  
Not sure, that might be the problem. 

file: [aggregate test.ifc.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/12654485/aggregate.test.ifc.txt)

https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/dae62b23-69e9-4bc2-bea5-d0b5bbc1d488

**Comment by brunoperdigao** [comment]
The `refresh_aggregate` use the same algorithm from the simple `shift+d` to duplicate the objects and recreate their relationships. Could you test this without aggregates by just duplicating the objects with `shift+d` and check if the problem still occurs?

**Comment by brunoperdigao** [comment]
Since I'll have to refactor the duplication process, I'll try to incorporate your [request](https://community.osarch.org/discussion/comment/17435/#Comment_17435)

**Comment by theoryshaw** [comment]
You're right, just Shift-d causes the same problem where the walls are assigned to multiple slabs. 


https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/0692a743-25cd-4f4f-85ec-41a5764ecc5c

**Comment by brunoperdigao** [comment]
I created a PR (#3789) to deal with the problem with connections. However, I couldn't find the cause of the original issue (walls missing after `shift+g`). I'll try to investigate your file (as far as my skills allow me to) because I couldn't reproduce the same behavior by testing in a new file.
